src/parser.c

```diff
@@ -848,10 +845,7 @@ enum parse_rv parse_ExportTransaction(struct ExportTransactionState *const state
     switch (state->state) {
         case 0: // ChainID
             CALL_SUBPARSER(id32State, Id32);
-            if(is_pchain_transaction(meta->type_id)) {
-              if(memcmp(network_info_from_network_id_not_null(meta->network_id)->x_blockchain_id, state->id32State.buf, sizeof(blockchain_id_t)))
-                REJECT("Invalid XChain ID");
-            } else {
+            if(!is_pchain_transaction(meta->type_id)) {
```

Comment:
I think this block should be unconditional. Now that I am more familiar with the code, I think the the old `else` was misleading and what is really going on is that that we were merely doing "else if got by assert" which could just as well apply to the rest of the function.